rocAL/source/api/rocal_api_data_loaders.cpp

```diff
@@ -2142,10 +2143,31 @@ rocalAudioFileSourceSingleShard(
                                                                                         context->master_graph->meta_data_reader()
                                                                                         );
         context->master_graph->set_loop(loop);
-        if(is_output) {
-            auto actual_output = context->master_graph->create_tensor(info, is_output);
-            context->master_graph->add_node<CopyNode>({output}, {actual_output});
-        }
+        if(downmix) {
+            TensorInfo output_info = info;
+            std::vector<size_t> output_dims = {context->user_batch_size(), info.dims()[1], 1};
```

Comment:
We can add one more condition for downmix stating
if max channels in the dataset is 1. Then we need not add downmix in the pipeline and just return the decoded output

in RPP node, when input channels is 1, downmix just copies input to output
https://github.com/ROCm/rpp/blob/master/src/modules/cpu/kernel/down_mixing.hpp#L52

So by checking in rocAL if max channels in dataset to 1 and not calling downmix for that case, we can just return the decoder output and avoid additional memcpy in RPP

Reply:
Also I would suggest one thing

Keep the output variable, If downmix node is the output then set the downmix tensor to the output variable

Add the if (is_output) condition with CopyNode only once at the end, Please check